internal/api/validator/validator.go

```diff
@@ -28,7 +28,7 @@ type Validator struct {
 func New(opts Options) *Validator {
 	jobNameSanitizer := opts.JobNameSanitizer
 	if jobNameSanitizer == nil {
-		jobNameSanitizer = strings.NewReplacer("_", "", ":", "", "-", "")
+		jobNameSanitizer = strings.NewReplacer("_", "", ":", "", "-", "", " ", "")
```

Comment:
IIRC the user had an actorID with a space in it. Does this successfully handle that and give the actorID back with a space in it? I imagine if we remove the space and give dapr back the actorID without the space that its going to error bc it wont find the actorID since the space was removed, unless we do some trimming in runtime and check that edge case in dapr?

Reply:
We are not actually removing the space character from the job name- we are only temporarily doing it here to sanitizes the string given to the DNS validator. The DNS validator doesn't allow space characters, but we still want to use that as a base policy, so we just sed replace the spaces which are given to it.

Check the tests added in cron_test.go - CRUD all still happens on the name with the space character.

Reply:
ah ok